app/src/main/java/com/example/babytracking/BreastActivity.kt

```diff
@@ -1,11 +1,134 @@
+import kotlinx.android.synthetic.main.breast_layout.*
+import kotlinx.android.synthetic.main.sleep_layout.*
+import java.sql.Timestamp
+import java.util.*
+
+/*
+This screen will allow to enter the data about the breastfeeding sessions of the baby.
+The first data required is the date and time when the session occurred.
+Those two are taken care of by the objects PickerSetup, which allow to pick the date and time,
+and TimeFormatter, which format the time so that we can read it.
+The other data that can be entered in this screen are the length of the session,
+at which breast did the baby drink, as well as how the session was for the baby and for the mother.
+*/
+class BreastActivity : AppCompatActivity() {
+
+    private var breast: String = ""
+    private var cal: Calendar = Calendar.getInstance()
+
 
-class BreastActivity: AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.breast_layout)
+
+        PickerSetup.setupDatePicker(this, breast_Date, cal)
+        PickerSetup.setupTimePicker(this, breast_Time, cal)
+        setupBreastSpinner()
+
+        breast_Save_Button.setOnClickListener {
+            saveInput()
+        }
+    }
+
+    private fun setupBreastSpinner() {
+
+        breast_Spinner ?: return
+        val breastSpinnerAdapter = ArrayAdapter.createFromResource(
+            this, R.array.array_breast,
+            android.R.layout.simple_spinner_item
+        )
+
+        breastSpinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        breast_Spinner.adapter = breastSpinnerAdapter
+
+
+        breast_Spinner.onItemSelectedListener =
+            object : AdapterView.OnItemSelectedListener {
+                override fun onItemSelected(
+                    parent: AdapterView<*>, view: View, pos: Int, id: Long
+                ) {
+                    if (!TextUtils.isEmpty(parent.getItemAtPosition(pos) as String)) {
+                        breast = breast_Spinner.selectedItem.toString()
+                    }
+                }
+
+                override fun onNothingSelected(parent: AdapterView<*>?) {
+                    breast = getString(R.string.unknown_breast)
+                }
+            }
+
+    }
+
+
+    @Suppress("DEPRECATION")
+    @SuppressLint("NewApi")
+    private fun saveInput() {
+
+        val lengthText: String = breast_Length.text.toString().trim()
+
+        if (lengthText.isEmpty()) {
+            breast_Length.error = "Length of breastfeeding required"
+            breast_Length.requestFocus()
+            return
+        }
+
+        val lengthInt: Int = lengthText.toInt()
+        val sessionQualForBabyText: String = breast_Quality_Baby.text.toString().trim()
+        val sessionQualForMotherText: String = breast_Quality_Mother.text.toString().trim()
+
+        val timestamp: Timestamp = TimeFormatter.makeTimeStamp(cal)
```

Comment:
This is very unclear: the calendar hides all the logic about which timestamp it is (and your variable name doesn't help). Could you store the timestamp in this class instead of a calendar?